src/main.rs

```diff
@@ -78,6 +78,34 @@ fn main_thread_checker_path() -> Result<Option<PathBuf>> {
     }
 }
 
+/// Get the specific path from the output of an external program.
+///
+/// E.g. `rustc --print sysroot` or `realpath $file`.
+fn get_external_path(mut cmd: Command, args: &[&str]) -> Result<PathBuf> {
+    let shell_command = format!(
+        "`{cmd} {args}`",
+        cmd = cmd.get_program().display(),
```

Comment:
Aww, this uses https://doc.rust-lang.org/std/ffi/struct.OsStr.html#method.display which is 1.87.

```suggestion
        cmd = cmd.get_program(). to_string_lossy(),
```

Reply:
Requiring 1.87 should be fine, we only support fairly recent nightlies anyway (but we avoid using nightly features in the code directly).

You can bump the version used for fmt+clippy.